irc/channelreg.go

```diff
@@ -63,6 +63,8 @@ type RegisteredChannel struct {
 	Invites map[string]MaskInfo
 	// Settings are the chanserv-modifiable settings
 	Settings ChannelSettings
+	// Metadata set using the METADATA command
+	Metadata MetadataStore
```

Comment:
Let's make this `map[string]string` --- that way if we redefine this type, the compiler will warn us of the incompatibility with the persisted data